src/ft2font_wrapper.cpp

```diff
@@ -381,115 +376,84 @@ static unsigned long read_from_file_callback(FT_Stream stream,
                                              unsigned char *buffer,
                                              unsigned long count)
 {
-
-    PyFT2Font *def = (PyFT2Font *)stream->descriptor.pointer;
-
-    if (fseek(def->fp, offset, SEEK_SET) == -1) {
-        return 0;
+    PyObject *py_file = ((PyFT2Font *)stream->descriptor.pointer)->py_file;
+    PyObject *seek_result = NULL, *read_result = NULL;
+    Py_ssize_t n_read = 0;
+    if (!(seek_result = PyObject_CallMethod(py_file, "seek", "k", offset))
+        || !(read_result = PyObject_CallMethod(py_file, "read", "k", count))) {
+        goto exit;
     }
-
-    if (count > 0) {
-        return fread(buffer, 1, count, def->fp);
+    char *tmpbuf;
+    if (PyBytes_AsStringAndSize(read_result, &tmpbuf, &n_read) == -1) {
+        goto exit;
     }
-
-    return 0;
+    memcpy(buffer, tmpbuf, n_read);
```

Comment:
Do you have any insights how often libpng calls this callback?

Reply:
My guess is just once to load the file, but no guarantees here.

Reply:
Looks like calls with count=0 are simply used, well, to seek() the file, but given that FT *only* interacts with the file via this callback, and that read offsets are always given from the file start, we could even just skip pure seek()s (i.e. immediately return if `count=0`) as later read()s will anyways pass an absolute offset.  Adding the early return doesn't seem to break anything, do you want that?